test/unit/test-navigation.js

```diff
@@ -118,6 +124,17 @@ describes.sandboxed('Navigation', {}, () => {
           expect(handleCustomProtocolSpy).to.be.calledWith(event, anchor);
         });
 
+        it('should select a custom linker target', () => {
+          const handleClickSpy = sandbox.spy(handler, 'handleClick_');
```

Comment:
I'm not familiar with this file, but both stubbing a private method and manually calling a private method are considered bad unit test patterns. They're testing the actual implementation and not the behavior, which reduces the tests coverage and makes the code harder to refactor.

Is there any way to improve this at this test level? Of course you don't have to refactor the whole file, I'm just wondering if there's a way to improve this `sandbox.spy`, not `handler.handle_` as it seems to be used across the file.
(Still LGTM)

Reply:
That's true, it's just the way I found to testing this type of behavior on this file. 

I changed it to use the previously defined `handleNavSpy` so I don't create another one like this, which is a little bit better.

Reply:
Good point. I think we can even just change this particular test to invoke the handler via the document used to get the navigation service. e.g document.click(). Add a TODO to fix this throughout the test.

Reply:
Sg, done.